### DISCUSSION TITLE
Transcribing numpy array.

### DISCUSSION THREAD

**Discussion by JenJenUwU** [SYSTRAN/faster-whisper#1133] [Q&A] [answered]
I am trying to create a real-time transcriber, and I would like to know if I can use the transcriber object directly on a numpy array containing my data, as I would like to remove the delay from OS operation. From the type of audio the transcribe function takes, it seems to be possible. However, when I take my buffer, which is defined by 

`buffer = np.zeros((sample_rate * seconds, 2))
`
However, I keep on receiving the error:

`numpy._core._exceptions._ArrayMemoryError: Unable to allocate 3.35 TiB for an array with shape (960000, 480002) and data type float64`, despite it being 10 seconds long with a sample rate of 48000. Is there any way I could format the array such that the function can read it?

**Comment by MahmoudAshraf97** [comment, marked as answer]
Transcribe accepts mono audio, the shape should be `(16000 * length_s)`
